File: 20260331_Contenido/20260331_Contenido_2_info.py
"""
Video: youtube.com/shorts/JH2xjFbUk_Q

Cómo determinar si una cadena es heterograma con Python:

Podemos crear una función personalizada. Por medio de comprensión de lista, el método lower (que es derivado de lowercase que significa minúsculas), y el método isalpha() (que signfica es alfa), obtenemos las letras de la cadena en minúsculas y las almacenamos en una variable. Creamos otra variable para un conjunto de la cadena, el cual obtiene las letras únicas. Comparamos la longitud de la cadena con la longitud del conjunto. Después imprimimos el resultado.
"""

def es_heterograma(cadena):
    letras = [caracter for caracter in
    cadena.lower() if caracter.isalpha()]
    letras_unicas = set(letras)
    return len(letras_unicas) == len(letras)

cadena1 = "república"
print(es_heterograma(cadena1))

File: 20260331_Contenido/20260331_Contenido_2_info.js
/*
Video: youtube.com/shorts/cV-VKnSLmrw

Cómo determinar si una cadena es heterograma con JavaScript:

Podemos crear una función personalizada. Creamos  una variable para almacenar los caracteres que obtenemos por medio del método toLowerCase(), que signfica a minúsculas. Usamos el método replace, que significa reemplazar, para eliminar los caracteres que no son letras y almacenamos el resultado en una variable. Creamos otra variable para almacenar un conjunto de la cadena, el cual obtiene las letras únicas. Comparamos la longitud de la cadena con la longitud del conjunto. Después mandamos el resultado a la consola.
*/

function esHeterograma(cadena) {
  let enMinusculas = cadena.toLowerCase();
  let letras = enMinusculas.replace(/[^a-z]/g, '');
  let letrasUnicas = new Set(letras);

  return letrasUnicas.size == letras.length;
}

let cadena1 = "república";
console.log(esHeterograma(cadena1));
